client/extensions/woocommerce/state/sites/settings/general/handlers.js

```diff
@@ -0,0 +1,53 @@
+/**
+ * Internal dependencies
+ */
+import { areSettingsGeneralLoaded } from 'woocommerce/state/sites/settings/general/selectors';
+import { dispatchRequest } from 'state/data-layer/wpcom-http/utils';
+import { http } from 'state/data-layer/wpcom-http/actions';
+import { setError } from '../../status/wc-api/actions';
+import {
+	WOOCOMMERCE_SETTINGS_GENERAL_REQUEST,
+	WOOCOMMERCE_SETTINGS_GENERAL_REQUEST_SUCCESS,
+} from 'woocommerce/state/action-types';
+
+export const handleSettingsGeneralSuccess = ( { dispatch }, action, next, { data } ) => {
+	const { siteId } = action;
+	dispatch( {
+		type: WOOCOMMERCE_SETTINGS_GENERAL_REQUEST_SUCCESS,
```

Comment:
> It's the action that is used in the reducer

I know but that's not enough reason to keep it that way :)

What bothers me is that the reducer handles `success` one way (it receives a specific action for that) but `error` in a totally different way (it needs to tap into the generic `WOOCOMMERCE_ERROR_SET` action and check if the original action was `SETTINGS_GENERAL_REQUEST`). I think both success and error should be handled in the same format, either with `SOMETHING_SUCCESS`/`SOMETHING_ERROR` action or with `GENERIC_SUCCESS`/`GENERIC_ERROR` actions.